plugins/database/mysql/connection_producer.go

```diff
@@ -110,6 +124,14 @@ func (c *mySQLConnectionProducer) Init(ctx context.Context, conf map[string]inte
 		mysql.RegisterTLSConfig(c.tlsConfigName, tlsConfig)
 	}
 
+	if c.RawConfig["auth_type"] == authTypeIAM {
```

Comment:
I think it'd be an improvement to lift the `auth_type` and `credentials` fields into the credential producer struct ([example](https://github.com/hashicorp/vault/blob/main/plugins/database/mysql/connection_producer.go#L26-L36)) instead of accessing them via the `RawConfig`. The mapstructure [decode](https://github.com/hashicorp/vault/blob/main/plugins/database/mysql/connection_producer.go#L59) is in place to populate them. This is the general pattern for fields that are a part of the database secrets engine's public API but are consumed by the database plugins.

Reply:
Thanks for the feedback! The parameters have now been exposed as part of the API in both cases (Postgres and MySQL)